modules/component/src/core/utils/has-zone.ts

```diff
@@ -14,5 +17,12 @@ import { NgZone } from '@angular/core';
  * ```
  */
 export function hasZone(z: NgZone): boolean {
-  return z.constructor.name !== 'NoopNgZone';
+  let isNgZone = false;
+
+  function fn() {}
+  fn.apply = () => (isNgZone = true);
+
+  z.runOutsideAngular(fn);
+
+  return isNgZone;
```

Comment:
This feels overly complex.

Would the check for `instanceof` work? Or are none of those classes exported?

Also, `Zone?.current?.name === 'angular'` might work. Needs a bit of testing, since this is just guessing right now.